services/auth/src/main/java/org/eclipse/hono/service/auth/impl/SimpleAuthenticationServer.java

```diff
@@ -110,17 +81,16 @@ protected void setRemoteConnectionOpenHandler(final ProtonConnection connection)
      * Checks if the open frame contains a desired <em>ADDRESS_AUTHZ</em> capability and if so,
      * adds the authenticated clients' authorities to the properties of the open frame sent
      * to the peer in response.
-     * 
+     *
      * @param connection The connection opened by the peer.
      */
     @Override
     protected void processRemoteOpen(final ProtonConnection connection) {
-        final boolean isAddressAuthz = Arrays.stream(connection.getRemoteDesiredCapabilities())
-                .anyMatch(symbol -> symbol.equals(CAPABILITY_ADDRESS_AUTHZ));
-        if (isAddressAuthz) {
+        final AddressAuthzProcessor addressAuthzProcessor = new AddressAuthzProcessor();
```

Comment:
Wondering as this class doesn't hold any state if this all could be just static calls?

Reply:
Right, I've changed the methods accordingly and renamed the class to `AddressAuthzHelper`.